test/GeneratorTests.cpp

```diff
@@ -23,3 +20,19 @@ TEST_F(GeneratorTests, checkConversionFromIntToFieldsEnum)
 
     ASSERT_EQ(result, ConvertIntToFieldEnum(input));
 }
+
+INSTANTIATE_TEST_CASE_P(Default,
+                        GeneratorParamTests,
+                        testing::Combine(testing::Range(3, 5),   // rows
```

Comment:
These comments are misleading. 3 is row, 5 is cols. Please remove comments and you can also add some more cases like 2, 2 or 5, 5

Reply:
In this part there are passed tuples<int, int> to board constructor. There ale genereted all cases for board col/row size from 3 to 4. Board of size 5x5 has 2^25 permutations and this is more than my computer has memory. Board with col/rows smaller then 3 are found to be invalid in JSONFileREader.